src/plugins.ts

```diff
@@ -0,0 +1,27 @@
+import * as ts from "typescript";
+import { Printer } from "./LuaPrinter";
+import { Visitors } from "./transformation/context";
+
+export interface Plugin {
+    /**
+     * An augmentation to the map of visitors that transform TypeScript AST to Lua AST.
+     *
+     * Key is a `SyntaxKind` of a processed node.
+     */
+    visitors?: Visitors;
+
+    /**
+     * A function that converts Lua AST to a string.
+     *
+     * Only one plugin can provide `printer`.
```

Comment:
It's supposed to mean that it can be provided either by 0 or 1 plugins passed to the the compilation. Any wording I can think of sounds a bit confusing, could you make a suggestion?

Reply:
I have some doubts with this entire mechanism. If this can only be supplied once, why give it as an option on all plugins. Would it be an idea to do something like `type Plugin = TransformerPlugin | PrinterPlugin`?